components/App.js

```diff
@@ -0,0 +1,80 @@
+var GIPHY_API_URL = 'http://api.giphy.com';
+var GIPHY_PUB_KEY = 'sLQchj92LgrtXIpE71K4aUeM5ec1emD4';
+
+App = React.createClass({
+    getInitialState() {
+        return {
+            loading: false,
+            searchingText: '',
+            gif: {}
+        };
+    },
+
+    handleSearch: function(searchingText) {
+        this.setState({
+          loading: true
+        });
+        var self = this;
+        this.getGif(searchingText)
+        .then(function(gif) {
+            self.setState({
+                loading: false,
+                gif: gif,
+                searchingText: searchingText
+              });
+            document.getElementById("errorMessage").innerText = "";
+        })
+        .catch(function(error) {
+            self.setState({
+                loading: false
+              });
+            document.getElementById("errorMessage").innerText = error.message;
```

Comment:
aj aj aj nie :P to jest React, nie kombinuj tu z realnym DOMem tylko zmień stan komponentu odpowiednio, a w `render` zależnie od tego stanu wyświetlaj informację z błędem